AOC3/src/main/kotlin/Day16.kt

```diff
@@ -15,19 +15,16 @@ fun main() {
     while (i <= iterations) {
         orders.forEach { order ->
             when (order[0]) {
-                's' -> spinDanceMove(Integer.parseInt(order.slice(1 until order.length)))
+                's' -> spinDanceMove(order.drop(1).toInt())
                 'x' -> {
-                    val orderExchangeKey = order.split("/")
-                    val firstPosition =
-                        Integer.parseInt(orderExchangeKey[0].slice(1 until orderExchangeKey[0].length))
-                    val secondPosition = Integer.parseInt(orderExchangeKey[1])
+                    val (firstPosition, secondPosition) = order.drop(1).split("/").map { it.toInt() }
                     exchangeDanceMove(firstPosition, secondPosition)
                 }
                 'p' -> partnerDanceMove(order[1], order[3])
                 else -> println("This $order is not known.")
             }
         }
-        if (i == 1) {
+        if (i == 1) { //

```

Comment:
I understand that these `if` statements were added as the solution for part 2, but they're awkward and don't read well. It's not really clear what's going on here.